octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/util/ClientLoginCookie.java

```diff
@@ -0,0 +1,104 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+package com.hpe.octane.ideplugins.eclipse.ui.util;
+
+import java.io.IOException;
+import java.net.HttpCookie;
+import java.util.List;
+import java.util.Optional;
+
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.hpe.adm.nga.sdk.authentication.SimpleUserAuthentication;
+import com.hpe.octane.ideplugins.eclipse.Activator;
+
+public class ClientLoginCookie {
+
+    private static HttpResponse httpResponse;
+    private static HttpCookie lwssoCookie;
+
+    public static HttpResponse loginClient(SimpleUserAuthentication user) {
+
+        HttpRequest httpRequest;
+        HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+        HttpRequestFactory requestFactory = HTTP_TRANSPORT.createRequestFactory();
+
+        ByteArrayContent content = ByteArrayContent.fromString("application/json", user.getAuthenticationString());
+
+        try {
+            httpRequest = requestFactory.buildPostRequest(new GenericUrl(Activator.getConnectionSettings().getBaseUrl() + "/authentication/sign_in"),
+                    content);
+            httpResponse = httpRequest.execute();
+        } catch (IOException e) {
+            //bad luck, you can't login
+            return null;
+        }
+        return httpResponse;
+
+    }
+
+    protected static HttpCookie setLwssoCookie() {
+        HttpResponse httpResponse = getHttpResponse();
+        List<String> strHPSSOCookieCsrf1 = httpResponse.getHeaders().getHeaderStringValues("Set-Cookie");
+
+        for (String strCookie : strHPSSOCookieCsrf1) {
+            List<HttpCookie> cookies = HttpCookie.parse(strCookie);
+            Optional<HttpCookie> lwssoCookieOp = cookies.stream().filter(a -> a.getName().equals("LWSSO_COOKIE_KEY")).findFirst();
+            if (lwssoCookieOp.isPresent()) {
+                lwssoCookie = lwssoCookieOp.get();
+                break;
+            } else {
+                return null;
+            }
+        }
+        return lwssoCookie;
+    }
+
+    public static HttpResponse getHttpResponse() {
+        return httpResponse;
+    }
+
+    public static boolean isUserLoggedIn() {
+        if (getHttpResponse() == null) {
+            return false;
+        }
+        return true;
+    }
+
+    public static HttpResponse getDataForImage(String pictureLink) {
+
+        HttpCookie lwssoCookieCopy;
+        if (lwssoCookie == null) {
+            lwssoCookieCopy = setLwssoCookie();
+        } else {
+            lwssoCookieCopy = lwssoCookie;
```

Comment:
@vladbenchea I'm not familiar with how the cookie mechanism works right now, but it seems that you are caching the cookie from the first session. Does this still work if octane timesout or the user switches the workspace?
Perhaps I'm missing something

Reply:
No, it's not. 
I discussed this with Tibi and Andras and thought of another way of doing this, by putting the whole getCookie and saveImage in another class in common